src/docfx/build/link/FileLinkMapBuilder.cs

```diff
@@ -11,16 +11,18 @@ internal class FileLinkMapBuilder
     private readonly DocumentProvider _documentProvider;
     private readonly MonikerProvider _monikerProvider;
     private readonly ContributionProvider _contributionProvider;
+    private readonly string _hostName;
 
     private readonly Scoped<ConcurrentHashSet<FileLinkItem>> _links = new();
 
     public FileLinkMapBuilder(
-        ErrorBuilder errors, DocumentProvider documentProvider, MonikerProvider monikerProvider, ContributionProvider contributionProvider)
+        ErrorBuilder errors, DocumentProvider documentProvider, MonikerProvider monikerProvider, ContributionProvider contributionProvider, string hostName)
     {
         _errors = errors;
         _documentProvider = documentProvider;
         _monikerProvider = monikerProvider;
         _contributionProvider = contributionProvider;
+        _hostName = hostName;
```

Comment:
The format of this `hostName` is like `learn.microsoft.com`, while broken link lib should receive a completed URL (like `https://learn.microsoft.com/azure/introduction`. So I suggest to add `https://` here.